File: samples/v2/php/tests/ContactsTest.php
<?php

namespace Keap\Sdk\Core\Tests;

use PHPUnit\Framework\TestCase;
use Keap\Core\V2\Api\ContactApi;
use Keap\Core\V2\Configuration;
use Keap\Core\V2\HeaderSelector;
use GuzzleHttp\Client;

class ContactsTest extends TestCase
{
    public function testListContactsSucceeds()
    {
        $apiKey = getenv('KEAP_REST_API_SERVICE_ACCESS_TOKEN');
        if (!$apiKey) {
            throw new \InvalidArgumentException('KEAP_REST_API_SERVICE_ACCESS_TOKEN is not set');
        }

        $config = new Configuration();
        $config->setAccessToken($apiKey);

        // Inline class inheriting from HeaderSelector
        $headerSelector = new class extends HeaderSelector {
            public function selectHeaders(array $accept, string $contentType, bool $isMultipart): array
            {
                // Call the parent method to retain original behavior
                $headers = parent::selectHeaders($accept, $contentType, $isMultipart);
                
                // Add the custom Authorization header
                $headers['Authorization'] = 'Bearer ' . getenv('KEAP_REST_API_SERVICE_ACCESS_TOKEN');
                return $headers;
            }
        };

        $client = new Client();
        $contactApi = new ContactApi($client, $config, $headerSelector);
        $response = $contactApi->listContactsUsingGET1();

        echo 'Found ' . count($response->getContacts()) . ' contacts' . PHP_EOL;
        foreach ($response->getContacts() as $contact) {
            $this->assertNotNull($contact->getId(), "Contact has no ID set");
        }
    }
}

File: samples/v2/java/src/test/java/com/keap/ContactsTest.java
package com.keap;

import com.keap.core.sdk.ApiClient;
import com.keap.core.sdk.ApiException;
import com.keap.core.sdk.client.ContactApi;
import com.keap.core.sdk.model.Contact;
import com.keap.core.sdk.model.ListContactsResponse;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class ContactsTest {

    @Test
    void listContactsSucceeds() throws ApiException {
        String apiKey = System.getenv("KEAP_REST_API_SERVICE_ACCESS_TOKEN");
        if (apiKey == null) {
            throw new IllegalStateException("KEAP_REST_API_SERVICE_ACCESS_TOKEN is not set");
        }

        ApiClient apiClient = new ApiClient();
        apiClient.setAccessTokenSupplier(() -> apiKey);

        ContactApi contactApi = new ContactApi(apiClient);

        Assertions.assertNotNull(contactApi, "contactApi should not be null");
        ListContactsResponse response = contactApi.listContactsUsingGET1(
          null, // List.of("preferred_name","create_time"),
          null,
          null, // "date_created ASC",
          null, // 100,
          null
        );

        System.out.println(response.getContacts().size());
        for (Contact contact : response.getContacts()) {
            Assertions.assertNotNull(contact.getId());
        }
    }
}
